README.md

```diff
@@ -124,6 +124,90 @@ It is possible to disable sending HTTP requests to the targets with the `-dry-ru
 cat etc/requests.jsonl | ./ripley -pace "30s@1" -dry-run
 ```
 
+`-strict` mode causes ripley to panic when encountering an error:
+
+```bash
+cat etc/requests.jsonl | ./ripley -strict
+```
+
+```JSON
+{
+  "statusCode": 0,
+  "latency": 0,
+  "request": {
+    "method": "PROPFIND",
+    "url": "http://localhost:8080/",
+    "body": "",
+    "timestamp": "2021-11-08T19:00:00.03Z",
+    "headers": null
+  },
+  "error": "invalid method: PROPFIND"
+}
+```
+
+```bash
+panic: invalid method: PROPFIND
+```
+
+If `-strict` isn't specified, ripley will print the problematic request, skip it and exit with a non zero exit code when it's done processing all requeests. Problematic requests aren't replayed.
+
+```bash
+cat etc/requests.jsonl | ./ripley -dry-run > /dev/null
+```
+```bash
+echo $? #=> 126
+```
+
+## ripleysort
+
+If the access logs that ripley replays are not strictly ordered by their timestamps, such as when combining access logs from several different sources, the `ripleysort` command can stream sort them, without loading all of the requests in memory. This is useful when sorting large datasets that won't fit in the host system's available memory. The `-bufferlen` flag can be used to configure `ripleysort`'s request buffer in to order to correctly sort certain datasets. In `-strict` mode, `ripleysort` will panic when it emits an out of order request. Knowing when `ripleysort` emits an out of order request is useful when tuning `-bufferlen`.
```

Comment:
> such as when combining access logs from several different sources

Interesting. If I have a file which combines logs from 2 apps (`N` - number of logs from app A, `M` - number of logs from app B), then I would need to set the buffer to at least `N+1`. Which is still better than reading the whole file to memory, but maybe it's still a lot?

I'm not a go expert and I haven't tried this, but maybe this could be an alternative solution, I'm not sure:
1. read the file line by line
1. when reading a line, save the value of the `timestamp`, the starting position of the line, and the byte length of the line to an index - which can be a map where the timestamp is the key, or something like that
1. once we finished reading the file, we have a way smaller map in memory than the original file
1. iterate over this map (while ordering it by key (timestamp) if go doesn't do that automatically), and use [File.ReadAt](https://pkg.go.dev/os#File.ReadAt) / [File.Seek](https://pkg.go.dev/os#File.Seek) to read from the original file from the positions which are saved in the map, and generate the new ordered logs file

I'm not sure whether:
- the generated index would be way that smaller - I assume it would be, as it's just a list of timestamp + position + length
- `File.ReadAt` is able to work efficiently (e.g. not loading in the whole file to memory when seeking)

But I think this could also work 🤷‍♂️ 

----
I assume you were thinking of the option of not letting logs from multiple apps to be placed in one file, and instead implement a smart merging algorithm which would read from multiple files at once, line by line and produce a merged file instead.

Reply:
This is how I ran it, @dios-david, with `paste` *interleaving* the requests from 2 files:

```bash
paste -d '\n' requests.jsonl requests-meta.jsonl | sed '/^$/d' | ./ripleysort -strict -bufferlen 10000 
```

Reply:
ah cool! I assumed they are sequential in the logs, starting with app A then app B, etc. in this case this is fine 👍